File: Dot examples/ASP Book Examples/Book Examples/Chapter15/VB/App_Code/OnSaleProduct.vb

Public Class ProductBase

    Private _price As Decimal

    Public Overridable Property Price() As Decimal
        Get
            Return _price
        End Get
        Set(ByVal Value As Decimal)
            _price = value
        End Set
    End Property

End Class


Public Class OnSaleProduct
    Inherits ProductBase

    Public Overrides Property Price() As Decimal
        Get
            Return MyBase.Price / 2
        End Get
        Set(ByVal Value As Decimal)
            MyBase.Price = value
        End Set
    End Property

End Class

File: Dot examples/ASP Book Examples/Book Examples/Chapter15/CS/App_Code/OnSaleProduct.cs
public class ProductBase
{
    private decimal _price;

    public virtual decimal Price
    {
        get { return _price; }
        set { _price = value; }
    }

}


public class OnSaleProduct : ProductBase
{

    override public decimal Price
    {
        get { return base.Price / 2; }
        set { base.Price = value; }
    }

}
